OpenRA.Mods.Common/Traits/Explodes.cs

```diff
@@ -51,7 +52,7 @@ public void Killed(Actor self, AttackInfo e)
 				return;
 
 			var warhead = e.Warhead as DamageWarhead;
-			if (info.DeathType != null && warhead != null && !warhead.DamageTypes.Overlaps(info.DeathType))
+			if (warhead != null && !warhead.DamageTypes.Overlaps(info.DeathType))
```

Comment:
`HashSet.Overlaps(...)` only return true if there is at least one element in the intersection ([HashSet(T).Overlaps Method (System.Collections.Generic)](https://msdn.microsoft.com/en-us/library/bb355623%28v=vs.110%29.aspx)). That is `HashSet.Overlaps(...)` is always false if either is empty (`.Count == 0`).